### ISSUE TITLE
not working wesnoth.wml_actions.endlevel T.result{} ?

### ISSUE THREAD

**Issue by vgaming** [wesnoth/wesnoth#2621]
Steps to repro:
* start wesnoth v1.13.11+dev (8309ba63c1-Modified)  (IDK why it says "modified", it's clean. Will look up later.)
* start any map (tested on Afterlife scrolling)
* open Lua console (enable debug, then press `)
* when side1 is active, execute this:
```
T = wesnoth.require("lua/helper.lua").set_wml_tag_metatable {}
wesnoth.wml_actions.endlevel { T.result{side=1, result="defeat"}, T.result {side=3, result="victory"} }
```

Expected at this point: "defeat" dialog

Actually: "victory" dialog

Note: didn't test this in online multiplayer games yet, but there may be bugs there too. Definitely need to do at some point.

**Comment by jyrkive** [comment]
For the record, the more simple

```lua
wesnoth.wml_actions.endlevel({result = "defeat"})
```

is known to work (or at least worked a month ago). A unit test uses it.

**Comment by gfgtdf** [comment]
just tested the code you gave and worked as it shoudl for me (1.13.11 official windows release)

**Comment by Vultraz** [comment]
Can repro in a local MP game.

**Comment by gfgtdf** [comment]
ok i think the misunderstanding here is that as soon as the game ends, it does not matter who's side was active. You get a victory dialog if at least one of the local human sides was victorious. This is independent of whether the game ended via [endlevel] or via the normal victory condition (all remaining living sides are allied). This is currently the expected behaviour. You can test the same thing in any local mp 2 player game just do a suicide attack with your leader turing your turn and you'll get a victory dialog.

**Comment by Vultraz** [comment]
So... this is not a bug?

**Comment by vgaming** [comment]
Confirming it works as expected in networked multiplayer game (one side is shown victorious, the other defeated).

So the only question is about local offline game.
If you make a suicidal move and lose versus e.g. another human player in hotseat mode, should you see "victorious" message or "defeated"? I'd vote for "defeated".
Also, may there be additional cases where it would still be preferable to see "victorious" dialog? (I don't see such cases yet.)

**Comment by gfgtdf** [comment]
> So... this is not a bug?

at least not a bug in endlevel one could argue that the behviour he expected in local games might make more sense though.

**Comment by vgaming** [comment]
I've removed the "Bug" label and added "Enhancement" because the current design was made consciously, at some point.

I'd still vote to change the current behavior tho.

**Comment by gfgtdf** [comment]
> If you make a suicidal move and lose versus e.g. another human player in hotseat mode

this was probably not the usecase that we had in mind when we implemented this, another case ist for example when you play a networked 2vs2 multiplayer game with 2 players where each side controles the 2 allied sides. In that case it makes sense to show the victory message when your 'team' wins.

**Comment by vgaming** [comment]
@gfgtdf of course, it would make sense to show the "end game" message corresponding to your team, not somebody else. Still, your example describes a different situation, not the one of this issue.

**Comment by gfgtdf** [comment]
> it would make sense to show the "end game" message corresponding to your team

this is also probably nontriovial to implement, in particular since the 'allied with' relation might be nontransitive

**Comment by vgaming** [comment]
@gfgtdf wait, something very important was missed in the issue description. The WML (or Lua) code actually _explicitely_ tells whether a particular side lost or won. It says:

> wesnoth.wml_actions.endlevel { T.result{side=1, result="defeat"},

So how we might end up showing "victory" dialog? I don't feel teams being non-transitive has anything to do with this particular case..

**Comment by vgaming** [comment]
So I'm asking for a use case under the following constraints:
* the Lua code explicitly states that side=1 is defeated
* it's side=1 -s turn now
* you have to show "victory" dialog anyways

I personally don't see any such case yet, which is why I ask to make a change here.

**Comment by gfgtdf** [comment]
ok i did a quick git blame and it seems like the 'report victory if one local player won independent on whose turn it currently is' has been around since the beginning (https://github.com/wesnoth/wesnoth/commit/4dae65c945fd20c906acc0c92101dbe46629d873 is the first commit found)

**Comment by jostephd** [comment]
So, if I understand correctly, the issue here is that in an MP game where the local client controls more than one side, a victory message is shown if any one of the local sides won.

Could we just show the victory/defeat messages of _all_ local sides? Or of the side whose turn it is? Showing a victory message if _another local side_ was victorious is not intuitive.
